staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -523,6 +530,13 @@ func (w *watchCache) WaitUntilFreshAndList(ctx context.Context, resourceVersion
 	return result, rv, index, err
 }
 
+func (w *watchCache) shouldRegisterAsWaitingForFresh(resourceVersion uint64) (bool, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) || !w.notFresh(resourceVersion) {
+		return false, nil
+	}
+	return etcdfeature.DefaultFeatureSupportChecker.Supports(storage.RequestWatchProgress)
```

Comment:
```suggestion
	return w.Supports(storage.RequestWatchProgress)
```

Reply:
`watchCache` doesn't have any storage or `Supports` on it. How it connected to cache?

Reply:
Sorry I meant `store`.

Reply:
Ups, it should be still possible as `watchCache` is constructed by `cacher` which has access to `storage`. You can just define a interface `SupportChecker` and pass it via constructor.

Reply:
Using `SupportChecker` directly will need client, i think we need to pass `storage` to `watchCache`.